### PR TITLE
Adding IMAGE_F_ROM_FIXED flag

### PR DESCRIPTION
**Pull request by de-nordic**
The series of commits adds IMAGE_F_ROM_FIXED flag, which has already been defined in imgtool and used in mcumgr, but has been missed in mcuboot, and code that uses the flag to skip application images that have been placed in incorrect slot.

### CODE REVIEW

**Review by nvlsianpu — COMMENTED** [review]
looks good to me

File: boot/bootutil/src/loader.c [hunk 1]
```diff
@@ -566,6 +566,36 @@ boot_version_cmp(const struct image_version *ver1,
...
+ *          does not match the slot address.
+ */
+static bool
+boot_rom_address_check(struct boot_loader_state *state, size_t slot)
+{
+    const struct image_header *hdr = boot_img_hdr(state, slot);
+    uint32_t f_off = boot_img_slot_off(state, slot);
+    if (hdr->ih_flags & IMAGE_F_ROM_FIXED && hdr->ih_load_addr != f_off) {
+        BOOT_LOG_WRN("Image in %s slot at 0x%x has been built for offset 0x%x"\
+		     ", skipping", slot == 0 ? "primary" : "secondary", f_off,
```
Comment (utzig) [inline boot/bootutil/src/loader.c:587]: There are tabs here, fix to use spaces.

File: boot/bootutil/src/loader.c [hunk 2]
```diff
@@ -566,6 +566,36 @@ boot_version_cmp(const struct image_version *ver1,
...
+ * and whether the slot starts at that address.
+ *
+ * @returns 0 if IMAGE_F_ROM_FIXED flag is not set;
+ *          0 if IMAGE_F_ROM_FIXED flag is set and ROM address specified in
+ *            header matches the slot address;
+ *          1 if IMF_F_ROM_FIXED flag is set but ROM address specified in header
+ *          does not match the slot address.
+ */
+static bool
+boot_rom_address_check(struct boot_loader_state *state, size_t slot)
```
Comment (utzig) [inline boot/bootutil/src/loader.c:581]: Couldn't this be moved to the same #ifdef block where `boot_select_or_erase` is defined?

**Review by utzig — CHANGES_REQUESTED** [review]

File: boot/bootutil/src/loader.c (hunk 1 above)
Comment (de-nordic) [reply boot/bootutil/src/loader.c:587]: Will fix.

File: boot/bootutil/src/loader.c (hunk 2 above)
Comment (de-nordic) [reply boot/bootutil/src/loader.c:581]: It used to, but that block is included for different reasons where this function is not used which causes compilation errors on unused static.

**Review by d3zd3z — APPROVED** [review]
I'm ok with the change. We will need to add this to the growing list of things implemented that we don't have tests for, though.

**Review by utzig — APPROVED** [review]

**Comment by de-nordic** [comment]
> 
> 
> I'm ok with the change. We will need to add this to the growing list of things implemented that we don't have tests for, though.

Seems that testing this flag should be my responsibility, I will write test for this, just need to get familiar with mcuboot test framework.

**Pull request merged**